ast_arginfo.h

```diff
@@ -34,6 +34,17 @@ ZEND_BEGIN_ARG_INFO_EX(arginfo_class_ast_Node___construct, 0, 0, 0)
 	ZEND_ARG_TYPE_INFO_WITH_DEFAULT_VALUE(0, lineno, IS_LONG, 1, "null")
 ZEND_END_ARG_INFO()
 
+ZEND_BEGIN_ARG_WITH_RETURN_TYPE_INFO_EX(arginfo_class_ast_Node_parseCode, 0, 2, IS_STATIC, 0)
```

Comment:
I think one possible solution is to redefine IS_STATIC to ast\\NODE in ast.c above the include - I forget if that would actually work

And continue using ast_arginfo.h instead of ast_legacy_arginfo.h

```c
#ifndef IS_STATIC
// Workaround for `parse*` methods on Node with static return types in php < 8.0
#define IS_STATIC ast\\Node
#endif
```